src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -501,6 +506,11 @@ public bool ContainsSyntaxTree(SyntaxTree syntaxTree)
 
         protected abstract bool CommonContainsSyntaxTree(SyntaxTree? syntaxTree);
 
+        /// <summary>
+        /// Optional semantic model provider for this compilation.
+        /// </summary>
+        internal readonly SemanticModelProvider? SemanticModelProvider;
```

Comment:
Why expose this as a field instead of a property?

Reply:
Because I blatantly copy pasted the previous member: https://github.com/dotnet/roslyn/blob/789a25bbb89a48e964d3b0fa951d02f72d36ee43/src/Compilers/Core/Portable/Compilation/Compilation.cs#L504-L507

I will switch both of these to be auto-props.